File: my_py_pkg/my_py_pkg/number_counter.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64
from example_interfaces.srv import SetBool


class NumberCounter(Node):
    def __init__(self):
        super().__init__('number_counter')

        self.count_ = 0
        self.publisher_number_count_ = self.create_publisher(Int64, 'number_count', 10)
        self.subscriber_number = self.create_subscription(Int64, 'number', self.callback_number_count, 10)
        self.service_ = self.create_service(SetBool, 'reset_counter', self.callback_reset_counter)

        self.get_logger().info('number_counter Started...')

    def callback_reset_counter(self, req: SetBool.Request, res: SetBool.Response) -> SetBool.Response:
        if req.data:
            self.count_ = 0

        self.get_logger().info('number_counter Reset...')
        res.success = True
        return res


    def callback_number_count(self, msg: Int64):
        res = Int64()

        self.count_ += msg.data
        res.data = self.count_

        self.publisher_number_count_.publish(res)


def main(args=None):
    rclpy.init(args=args)
    node = NumberCounter()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: my_cpp_pkg/src/number_system/number_counter.cpp
#include <rclcpp/rclcpp.hpp>
#include <example_interfaces/srv/set_bool.hpp>
#include <example_interfaces/msg/int64.hpp>
#include <example_interfaces/msg/string.hpp>

using namespace std::chrono_literals;
using std::placeholders::_1;

class NumberCounter : public rclcpp::Node
{
private:
    rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr subscriber;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher;
    rclcpp::Service<example_interfaces::srv::SetBool>::SharedPtr service;
    int initial;
    int count;

    void numberCallback(const example_interfaces::msg::Int64::SharedPtr msg)
    {

        auto sender_msg = example_interfaces::msg::Int64();
        sender_msg.data = count += msg->data;
        publisher->publish(sender_msg);
    }

    void resetCounter(const std::shared_ptr<example_interfaces::srv::SetBool::Request> req, std::shared_ptr<example_interfaces::srv::SetBool::Response> res)
    {
        if (!req->data)
            return;

        res->success = true;
        count = initial;

        RCLCPP_INFO(this->get_logger(), "Reseted counter...");
    }

public:
    NumberCounter() : Node("number_count")
    {
        this->declare_parameter("initial_value", 5);
        initial = count = this->get_parameter("initial_value").as_int();

        subscriber = this->create_subscription<example_interfaces::msg::Int64>("number", 10, std::bind(&NumberCounter::numberCallback, this, _1));
        publisher = this->create_publisher<example_interfaces::msg::Int64>("number_count", 10);
        service = this->create_service<example_interfaces::srv::SetBool>("reset_counter", std::bind(&NumberCounter::resetCounter, this, std::placeholders::_1, std::placeholders::_2));

        this->declare_parameter("test123", 5);
        RCLCPP_INFO(this->get_logger(), "number_count Started...");
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    rclcpp::spin(std::make_shared<NumberCounter>());

    rclcpp::shutdown();
    return 0;
}